core/controllers/blog_dashboard.py

```diff
@@ -353,3 +355,120 @@ def delete(self, blog_post_id: str) -> None:
         blog_domain.BlogPost.require_valid_blog_post_id(blog_post_id)
         blog_services.delete_blog_post(blog_post_id)
         self.render_json(self.values)
+
+
+class BlogDashboardBlogPostStatisticsHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Handler for blog dashboard statistics tab for blog post."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'blog_post_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        },
+        'chart_type': {
+            'schema': {
+                'type': 'basestring',
+                'choices': [
+                    'views',
+                    'reads',
+                    'reading_time'
+                ]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_access_blog_dashboard
+    def get(self, blog_post_id: str, chart_type: str) -> None:
+        """Populates the data for generating statistics plot."""
+
+        if chart_type == 'views':
+            stats = blog_statistics_domain.BlogPostViewsAggregatedStats
+            stats = blog_statistics_services.get_blog_post_views_stats_by_id(
+                blog_post_id
+            )
+        elif chart_type == 'reads':
+            stats: blog_statistics_domain.BlogPostReadsAggregatedStats
```

Comment:
This is not a legal pattern, remove this!

Reply:
Removed!